test/src/test/java/org/corfudb/integration/AbstractIT.java

```diff
@@ -225,41 +225,37 @@ public static void waitFor(Supplier<Boolean> supplier) throws InterruptedExcepti
      *
      * @param pid parent process identifier
      * @return list of children process identifiers
-     *
-     * @throws IOException
+     * @throws IOException {@inheritDoc}
+     * @throws InterruptedException {@inheritDoc}
      */
-    private static List<Long> getChildPIDs (long pid) {
+    private static List<Long> getChildPIDs (long pid) throws IOException, InterruptedException {
         List<Long> childPIDs = new ArrayList<>();
-        try {
-            // Get child pid(s)
-            ProcessBuilder builder = new ProcessBuilder();
-            builder.command("sh", "-c", "pgrep -P " + pid);
-            Process p = builder.start();
-            p.waitFor();
-
-            // Read output
-            BufferedReader br = new BufferedReader(new InputStreamReader(p.getInputStream()));
-            String line = null;
-            String previous = null;
-            while ((line = br.readLine()) != null) {
-                if (!line.equals(previous)) {
-                    previous = line;
-                    long childPID = Long.parseLong(line);
-                    childPIDs.add(childPID);
-                }
-            }
+        // Get child pid(s)
+        ProcessBuilder builder = new ProcessBuilder();
+        builder.command("sh", "-c", "pgrep -P " + pid);
+        Process p = builder.start();
+        p.waitFor();
```

Comment:
Can we get rid of this recursion?

`pkill -9 -P {pid}` would match all child process with a parent pid, why isn't that sufficient ?

Reply:
Sorry what does the `-9` mean? I didn't found this option.
For `-P` I thought only processes created by another process are considered to be the child process, so we may need to traverse the tree to get all child/grandchild processes.

Reply:
Oh it's the SIGKILL, I got you know. Thanks!

Reply:
`AbstractIT` launches the server by invoking the bash script `bin/corfu_server`. This script will start a process with the JVM. 

```
ps aux | grep corfu
maithem   243794  0.0  0.0   9632  3652 pts/1    S+   11:02   0:00 bash ./bin/corfu_server -ms 9000
maithem   243802 16.0  0.4 5863460 155444 pts/1  Sl+  11:02   0:01 /usr/lib/jvm/java-8-openjdk-amd64/bin/java -cp /home/maithem/CorfuDB/scripts/../infrastructure/target/infrastructure-0.3.1-SNAPSHOT-shaded.jar -Xmx2000m org.corfudb.infrastructure.CorfuServer -ms 9000

```
`pkill -9 -P {pid}` will kill the parent bash script and all child processes that have the same parent pid, so you don't need to implement the recursion here.

Reply:
Can you just confirm that for LR it kills all processes? I remember I had to add this cause there was no way the server got killed by killing just the parent process (but it could be that there was something else interfering at that time)